jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -301,20 +305,63 @@ public JibContainerBuilder addLabel(String key, String value) {
     return this;
   }
 
-  @VisibleForTesting
-  BuildConfiguration.Builder toBuildConfiguration(TargetImage targetImage) {
-    BuildConfiguration.Builder buildConfigurationBuilder = BuildConfiguration.builder();
+  /**
+   * Builds the container(s).
+   *
+   * @param containerizer the {@link Containerizer} that configures how to containerize
+   * @return the built container(s)
+   * @throws CacheDirectoryCreationException if a directory to be used for the cache could not be
+   *     created
+   * @throws ExecutionException if an exception occurred during execution
+   * @throws InterruptedException if the execution was interrupted
+   * @throws IOException if an I/O exception occurs
+   */
+  public JibContainer containerize(Containerizer containerizer)
+      throws InterruptedException, ExecutionException, IOException,
+          CacheDirectoryCreationException {
+    BuildConfiguration buildConfiguration =
+        toBuildConfiguration(BuildConfiguration.builder(), containerizer);
+    DescriptorDigest imageDigest =
+        containerizer.getTargetImage().toBuildSteps(buildConfiguration).run();
 
+    return new JibContainer(imageDigest);
+  }
+
+  /**
+   * Builds a {@link BuildConfiguration} using this and a {@link Containerizer}.
+   *
+   * @param buildConfigurationBuilder the {@link BuildConfiguration.Builder} to use
+   * @param containerizer the {@link Containerizer}
+   * @return the {@link BuildConfiguration}
+   * @throws CacheDirectoryCreationException if a cache directory could not be created
+   * @throws IOException if an I/O exception occurs
+   */
+  @VisibleForTesting
+  BuildConfiguration toBuildConfiguration(
```

Comment:
This is a private method used only be `containerize()`. Do we have to accept `buildConfigurationBuilder`? Can't this method just create it with `BuildConfiguration.builder()`?

Reply:
This is mostly for the test to be able to pass a `BuildConfiguration.Builder` to verify.

Reply:
I see, that makes sense. Just wondering if you can't directly examine the built `BuildConfiguration` using getters.

Reply:
Yea, it was mostly for the two caches that get generated on `BuildConfiguration.Builder#build` and didn't want to add a getter and new field to `Cache` just for this test.